demos/interactive_face_detection_demo/main.cpp

```diff
@@ -289,7 +287,7 @@ int main(int argc, char *argv[]) {
                     float intensity_mean = calcMean(prev_frame(rect));
 
                     if ((face == nullptr) ||
-                        ((face != nullptr) && ((std::abs(intensity_mean - face->_intensity_mean) / face->_intensity_mean) > 0.07f))) {
+                        ((std::abs(intensity_mean - face->_intensity_mean) / face->_intensity_mean) > 0.07f)) {
```

Comment:
Why was it removed? The behavior for `operator->` is undefined if the stored pointer is null.

Reply:
The second subexpression will only be evaluated if the first subexpression is false, and thus `face != nullptr`.